Mono.Debugging/Mono.Debugging.Evaluation/NRefactoryExpressionEvaluatorVisitor.cs

```diff
@@ -432,47 +434,49 @@ ValueReference EvaluateBinaryOperatorExpression (BinaryOperatorType op, ValueRef
 			return LiteralValueReference.CreateObjectLiteral (ctx, expression, res);
 		}
 
-		static string ResolveType (EvaluationContext ctx, TypeReferenceExpression mre, List<object> args)
-		{
-			var memberType = mre.Type as MemberType;
+		//static string ResolveType (EvaluationContext ctx, TypeReferenceExpression mre, List<object> args)
+		//{
+		//	var memberType = mre.Type as MemberType;
 
-			if (memberType != null) {
-				var name = memberType.MemberName;
+		//	if (memberType != null) {
+		//		var name = memberType.MemberName;
 
-				if (memberType.TypeArguments.Count > 0) {
-					name += "`" + memberType.TypeArguments.Count;
+		//		if (memberType.TypeArguments.Count > 0) {
+		//			name += "`" + memberType.TypeArguments.Count;
 
-					foreach (var arg in memberType.TypeArguments) {
-						var resolved = arg.Resolve (ctx);
+		//			foreach (var arg in memberType.TypeArguments) {
+		//				var resolved = arg.Resolve (ctx);
 
-						if (resolved == null)
-							return null;
+		//				if (resolved == null)
+		//					return null;
 
-						args.Add (resolved);
-					}
-				}
+		//				args.Add (resolved);
+		//			}
+		//		}
 
-				return name;
-			}
+		//		return name;
+		//	}
 
-			return mre.ToString ();
-		}
+		//	return mre.ToString ();
+		//}
 
-		static string ResolveType (EvaluationContext ctx, MemberReferenceExpression mre, List<object> args)
+		static string ResolveType (EvaluationContext ctx, MemberAccessExpressionSyntax mre, List<object> args)
 		{
 			string parent, name;
 
-			if (mre.Target is MemberReferenceExpression) {
-				parent = ResolveType (ctx, (MemberReferenceExpression) mre.Target, args);
-			} else if (mre.Target is TypeReferenceExpression) {
+			if (mre.Expression is MemberAccessExpressionSyntax mae) {
+				parent = ResolveType (ctx, mae, args);
+			} else /* TODO?
```

Comment:
All the type resolving is gone?

Reply:
TypeReferenceExpression is gone.

Reply:
resolving a type works now the same way as before, although I have to say I don't really see the point in it.

What's the point of resolving type "System.Console" or namespace "System"

Resolving instances of a type on the other hand makes perfect sense ........

Reply:
also, `typeof(System.Console)` is perfectly fine too.

Just not sure what the point of `System.Console` is

Reply:
@nosami `System.Console` should give me static members in that class.

Reply:
OK. I guess that does make sense.

I still don't think we should have namespaces though. In fact, the old code used to guess if something was a namespace https://github.com/mono/debugger-libs/blob/f7085b2d81cdf1acb85189428c149ac5b98b6bbc/Mono.Debugging/Mono.Debugging.Evaluation/NRefactoryExpressionEvaluatorVisitor.cs#L1158

Reply:
Yeah, namespaces make no sense. Static member sof a type do, though.

Reply:
This is all working now as before.